woocommerce/woocommerce-ios
# Shipping Labels: local validation of address fields

Part of #2973 

## Description
As part of the Shipping Labels M2, we should validate some fields of the Shipping Label Address also locally, in particular, we want to make sure that the user can't proceed without adding some fields that are mandatory, like name, address, city, country, ecc...
I fixed also an issue with the color of the activity indicator that substitute the "Done" button in the navigation bar when pressed. 

## Changes
- `Row` now conform to `Equatable` because we want to manage different type of errors in `fieldError` case.
- The `validateAddress` method allow us to validate an address locally or both locally + remotely.
- Refactor of how `sections` are managed and refreshed. Before, we built sections on the fly, when they were requested by the view controller. This could cause a misalignment with the state of the view model. Sections are now updated when the `State` undergoes a change.
- Updated unit tests.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell.
5. Go to the editing address screen, and try to delete one of the mandatory fields (like `name`). Press the Done button. -> You should see an error under the name field. 
6. The same things should happen pressing the button `Use address as Entered`, with the only difference that the address will be validated only locally, checking that the mandatory fields are fulfilled.

## Screenshots
<img src="https://user-images.githubusercontent.com/495617/111469909-8b91e200-8727-11eb-80ea-39289dc4a09b.png" width=300 />


Update release notes:
